ebmlite/util.py

```diff
@@ -20,11 +20,95 @@
 
 import ast
 from base64 import b64encode, b64decode
+from io import BytesIO
+import struct
 import sys
 import tempfile
 from xml.etree import ElementTree as ET
 
-from . import core, encoding
+from . import core, encoding, decoding
+
+# ==============================================================================
+#
+# ==============================================================================
+
+
+def createID(schema, idClass, exclude=(), min=0x80, max=0x1FFFFFFF, count=1):
+    """ Generate unique EBML IDs. Primarily intended for use 'offline' by
+        humans creating EBML schemata.
+
+        @param schema: The `Schema` in which the new IDs must coexist.
+        @param idClass: The EBML class of ID, one of (case-insensitive):
+            * `'a'`: Class A (1 octet, base 0x8X)
+            * `'b'`: Class B (2 octets, base 0x4000)
+            * `'c'`: Class C (3 octets, base 0x200000)
+            * `'d'`: Class D (4 octets, base 0x10000000)
+        @param exclude: A list of additional IDs to avoid.
+        @param min: The minimum ID value, within the ID class' range.
+        @param max: The maximum ID value, within the ID class' range.
+        @param count: The maximum number of IDs to generate. The result may be
+            fewer than specified if too few meet the given criteria.
+        @return: A list of EBML IDs that match the given criteria.
+    """
+    ranges = dict(A=(0x80, 0xFF),
+                  B=(0x4000, 0x7FFF),
+                  C=(0x200000, 0x3FFFFF),
+                  D=(0x10000000, 0x1FFFFFFF))
+    idc = idClass.upper()
+    if idc not in ranges:
+        raise KeyError('Invalid ID class %r: must be one of %r' %
+                       (idClass, list(ranges)))
+
+    # Keep range within the one specified and the one imposed by the ID class
+    idrange = (__builtins__['max'](ranges[idc][0], min),
+               __builtins__['min'](ranges[idc][1], max))
```

Comment:
this is... an interesting choice, that I somehow didn't notice the first time.

I'd personally prefer just renaming `min` -> `minId` or `minValue` or something (and respectively the same for `max`) to avoid this kind of odd method of resolving shadowed object names... but I guess it's fine?